.github/workflows/build.yml

```diff
@@ -2,132 +2,219 @@ name: Build
+    name: Build in GCC
+    runs-on: ubuntu-20.04
+    container:
+      image: ${{ matrix.container }}
+      options: --user root
+    strategy:
+      matrix:
+        container: [ "debian:11.1" ]
+        version: [ "9", "10" ] # gcc 11 doesn't exist in debian apt
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Compiler
         run: |
-          brew install cmake ninja
-          if [ "${{ matrix.compiler }}" = "gcc" ]; then
-            brew install gcc@${{ matrix.version }}
-            echo "CC=gcc-${{ matrix.version }}" >> $GITHUB_ENV
-            echo "CXX=g++-${{ matrix.version }}" >> $GITHUB_ENV
-          else
-            sudo xcode-select -switch /Applications/Xcode_${{ matrix.version }}.app
-            echo "CC=clang" >> $GITHUB_ENV
-            echo "CXX=clang++" >> $GITHUB_ENV
-          fi
+          apt-get update
+          apt-get install -y g++-${{ matrix.version }}
+          echo "CC=gcc-${{ matrix.version }}" >> $GITHUB_ENV
+          echo "CXX=g++-${{ matrix.version }}" >> $GITHUB_ENV
       - name: Install and Test
         run: (./install.sh && make tests)
 
-  build-external:
+  build-external-apt-clang:
     timeout-minutes: 5
-    name: Build External
+    name: Build in Clang
     runs-on: ubuntu-20.04
     container:
       image: ${{ matrix.container }}
       options: --user root
     strategy:
       matrix:
-        name: [
-            debian-stable-gcc-9,
-            debian-stable-gcc-10,
-            debian-stable-clang-9,
-            debian-stable-clang-11,
-        ]
-        include:
-          - name: debian-stable-gcc-9
-            container: debian:stable
-            compiler: gcc
-            version: "9"
+        container: [ "debian:11.1" ]
+        version: [ "9", "11" ] # clang 10 doesn't exist in debian apt
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Compiler
+        run: |
+          apt-get update
+          apt-get install -y clang-${{ matrix.version }}
+          echo "CC=clang-${{ matrix.version }}" >> $GITHUB_ENV
+          echo "CXX=clang++-${{ matrix.version }}" >> $GITHUB_ENV
+      - name: Install and Test
+        run: (./install.sh && make tests)
 
-          - name: debian-stable-gcc-10
-            container: debian:stable
-            compiler: gcc
-            version: "10"
+  build-external-yum-gcc:
+    timeout-minutes: 10
+    name: Build in default GCC
+    runs-on: ubuntu-20.04
+    container:
+      image: ${{ matrix.container }}
+      options: --user root
+    strategy:
+      matrix:
+        container: [ "fedora:34", "fedora:35", "amazonlinux:2", 'centos:8' ]
+    steps:
+      - name: Install AmazonLinux Deps
+        if: startsWith(matrix.container, 'amazonlinux')
+        run: yum -y install tar xz gzip
+      - name: Install CentOS Deps
+        if: startsWith(matrix.container, 'centos:8')
+        run: |
+          yum -y install epel-release dnf-plugins-core
+          yum config-manager --set-enabled powertools
```

Comment:
This must be in `install.sh`. How people would otherwise be able to compile on these platforms?

Reply:
This is specific to centos 8 in order to allow the sources to install the compiler. Since installing of the compiler is also not part of install.sh, this should also not be there.